Calculator/Calculator/Model/Calculator.swift

```diff
@@ -4,59 +4,159 @@ enum CalculatorError: Error {
     case dividedByZero
     case invalidNumber
     case invalidOperator
+    case noResult
+    
+    var errorMessage: String {
+        switch self {
+        case .dividedByZero:
+            return "NaN"
+        default:
+            return "Error발생"
+        }
+    }
 }
 
 class Calculator {
-    var numberStack = Stack<String>()
-    var result = ""
-    var postfix = Postfix()
+    private var infix: [String] = []
+    private var postfix: [String] = []
+    private var numberStack = Stack<String>()
+    private var operatorsStack = Stack<String>()
+    
+    func changeToPostfix(from infix: Array<String>) {
+        infix.forEach {
+            if isOperator(item: $0) {
+                addToPostfix(item: $0)
+            } else if operatorsStack.isEmpty() {
+                addToOperatorStack(item: $0)
+            } else {
+                comparePriority(with: changeOperatorType($0))
+            }
+        }
+        processLeftOperatorStack()
+    }
     
-    func isNumberInPostfix(item: String) -> Bool {
+    private func isOperator(item: String) -> Bool {
         let operators = ["+", "−", "×", "÷"]
-        
         return !operators.contains(item)
     }
     
-    func checkDividedNumber(second: Double) throws {
+    private func addToOperatorStack(item: String) {
+        operatorsStack.push(item: item)
+    }
+    
+    private func addToPostfix(item: String) {
+        postfix.append(item)
+    }
+    
+    private func comparePriority(with input: Operator?) {
+        guard let unwrappedLastOperator = operatorsStack.top() else { return }
+        let lastOfOperatorStack = changeOperatorType(unwrappedLastOperator)
+        guard let unwrappedLastOfOperatorStack = lastOfOperatorStack, let unwrappedInput = input else { return }
+        if unwrappedLastOfOperatorStack < unwrappedInput {
+            addToOperatorStack(item: unwrappedInput.operatorSymbol)
+        } else if unwrappedLastOfOperatorStack == unwrappedInput {
+            addToPostfix(item: unwrappedLastOfOperatorStack.operatorSymbol)
+            operatorsStack.pop()
+            addToOperatorStack(item: unwrappedInput.operatorSymbol)
+        } else {
+            processLeftOperatorStack()
+            addToOperatorStack(item: unwrappedInput.operatorSymbol)
+        }
+    }
+    
+    private func changeOperatorType(_ item: String) -> Operator? {
+        let lastStackOperator = item
+        
+        switch lastStackOperator {
+        case "+":
+            return Operator.plus
+        case "−":
+            return Operator.minus
+        case "×":
+            return Operator.multiply
+        case "÷":
+            return Operator.divide
+        default:
+            return nil
+        }
+    }
```

Comment:
이렇게 바꿔도 가독성이 좋을듯 하네요.
```suggestion
    private func convert(_ item: String) -> Operator? {
        let lastStackOperator = item
        
        switch lastStackOperator {
        case "+":
            return Operator.plus
        case "−":
            return Operator.minus
        case "×":
            return Operator.multiply
        case "÷":
            return Operator.divide
        default:
            return nil
        }
    }
```